subsys/net/lib/lwm2m/lwm2m_object.h

```diff
@@ -278,6 +278,7 @@ struct lwm2m_engine_context {
 	struct lwm2m_input_context *in;
 	struct lwm2m_output_context *out;
 	struct lwm2m_obj_path *path;
+	size_t block_offset;
```

Comment:
I'm curious why have the block_offset here in the engine_context as opposed to placing it in the block_context structure where it can be directly related to the struct zoap_block_context?  Feels like this should be moved.